website/docs/plugins/optuna_sweeper.md

```diff
@@ -114,8 +127,8 @@ The output is as follows:
 ```commandline
 [HYDRA] Study name: sphere
 [HYDRA] Storage: None
-[HYDRA] Sampler: TPESampler
-[HYDRA] Direction: minimize
+[HYDRA] Sampler: <optuna.samplers._tpe.sampler.TPESampler object at 0x7fd7901d6d30>
```

Comment:
@toshihikoyanase, it would be nice to have an informative string representation of the samplers.

Reply:
This used to print the enum, which was interpretable. That is gone now, Optuna doesn't have a nice repr for their samplers which causes this. I'm not sure what the best solution is to get a nice print out

Reply:
I ended up using `type(sampler).__name_`

Reply:
I think it should be straight forward and useful to add a `__repr__` to the samplers (and possibly to other objects).
Printing the type is good enough for now, if Optuna adds the repr we can use it here for more informative representation.

What is interesting to me is that by using recursive instantiation, we have lost the original config and now we have the object.
You can still gain access to the Hydra config though!
See https://github.com/facebookresearch/hydra/blob/master/examples/configure_hydra/job_name/no_config_file_override.py

Try something like:
`HydraConfig.get().sweeper.sampler`.
That should have the DictConfig object. if you print it as it will look like a plain dictionary.

Reply:
Sorry for the delayed response.

> I ended up using `type(sampler).__name__`
> I think it should be straight forward and useful to add a `__repr__` to the samplers (and possibly to other objects).

Thank you for your discussion. I totally agree with you and create a feature request in the Optuna repository (https://github.com/optuna/optuna/issues/2536).